rio_tiler/sentinel2.py

```diff
@@ -100,34 +142,67 @@ def _sentinel_parse_scene_id(sceneid):
     day = meta["acquisitionDay"].lstrip("0")
     img_num = meta["num"]
 
-    meta["key"] = "tiles/{}/{}/{}/{}/{}/{}/{}".format(
+    meta["scene"] = sceneid
+    meta["aws_bucket"] = AWS_SENTINEL_BUCKET + meta["processingLevel"].lower()
+    meta["aws_prefix"] = "tiles/{}/{}/{}/{}/{}/{}/{}".format(
         utm_zone, latitude_band, grid_square, year, month, day, img_num
     )
-
-    meta["scene"] = sceneid
+    meta["key"] = meta["aws_prefix"]  # Will be deprecated in rio-tiler v2.0.0
+
+    if meta["processingLevel"] == "L1C":
+        meta["preview_file"] = "preview.jp2"
+        meta["preview_prefix"] = "preview"
+        meta["bands"] = list(
+            itertools.chain.from_iterable(
+                [bands for _, bands in SENTINEL_L1_BANDS.items()]
+            )
+        )
+        meta["valid_bands"] = meta["bands"]
+    else:
+        meta["preview_file"] = "R60m/TCI.jp2"
+        meta["preview_prefix"] = "R60m"
+        meta["bands"] = SENTINEL_L2_BANDS["60m"]
+        meta["valid_bands"] = meta["bands"] + SENTINEL_L2_PRODUCTS["60m"]
 
     return meta
 
 
+def _l2_prefixed_band(band):
+    """Return L2A prefixed bands name."""
+    if band in SENTINEL_L2_BANDS["60m"]:
+        for res, bands in SENTINEL_L2_BANDS.items():
```

Comment:
This strategy relies on the order of key/value pairs in the dictionary to find the highest resolution for each band. Based on some quick research, it appears Python versions prior to 3.7 did not track dictionary insertion order, and therefore the order in which elements are returned from iterators like `.items()` cannot be relied upon. Unless I'm reading the docs wrong, which is entirely possible 😂

Reply:
Oh no I think you are right ;-)
I'll try to come up with tests and use https://docs.python.org/2/library/collections.html#collections.OrderedDict